### PR TITLE
Inject and configure the Develocity extension

### PR DESCRIPTION
**Pull request by pshevche**
## Summary

Reviewing commit-by-commit should be pretty straight-forward. The Maven template has been updated to:
1. Download the Develocity/Gradle Enterprise extension based on the version specified.
2. Inject the downloaded extension only if neither custom nor Develocity nor GE extension is applied.
3. Configure both the Develocity and GE extensions.

### CODE REVIEW

File: develocity-maven.yml [hunk 1]
```diff
@@ -41,6 +41,16 @@ spec:
...
+  
+  function isAtLeast() {
+    local v1=$(printf "%03d%03d%03d%03d" $(echo "$1" | cut -f1 -d"-" | tr '.' ' '));
+    local v2=$(printf "%03d%03d%03d%03d" $(echo "$2" | cut -f1 -d"-" | tr '.' ' '));
+    if [ "$v1" -eq "$v2" ] || [ "$v1" -gt "$v2" ]; then
+      echo "true";
+    else
+      echo "false";
+    fi
+  }
```
Comment (alextu) [inline develocity-maven.yml]: You can get rid of the `;`, see:
```suggestion
  function isAtLeast() {
    local v1=$(printf "%03d%03d%03d%03d" $(echo "$1" | cut -f1 -d"-" | tr '.' ' '))
    local v2=$(printf "%03d%03d%03d%03d" $(echo "$2" | cut -f1 -d"-" | tr '.' ' '))
    if [ "$v1" -eq "$v2" ] || [ "$v1" -gt "$v2" ]
    then
      echo "true"
    else
      echo "false"
    fi
  }
```

File: develocity-maven.yml [hunk 2]
```diff
@@ -80,22 +96,26 @@ spec:
...
   }
-
-  function dvCoordinates() {
-    local coordinates='com.gradle:gradle-enterprise-maven-extension'
+  
+  function detectDvExtension() {
     if [ ! -z "${customMavenExtensionCoordinates}" ]
     then
-      coordinates="${customMavenExtensionCoordinates}"
+      echo "$(detectExtension $1 $customMavenExtensionCoordinates)"
```
Comment (alextu) [inline develocity-maven.yml]: I would assign a more explicit name to `$1` (`rootDir` ?) to have a bit more context

File: develocity-maven.yml [hunk 3]
```diff
@@ -80,22 +96,26 @@ spec:
...
+      local appliedDefaultExtension="$(detectExtension $1 'com.gradle:gradle-enterprise-maven-extension')"
+      if [ "${appliedDefaultExtension}" = "false" ]; then
+        appliedDefaultExtension="$(detectExtension $1 'com.gradle:develocity-maven-extension')"
+      fi
+      echo "${appliedDefaultExtension}"
     fi
-    echo "${coordinates}"
   }
 
   function ccudCoordinates() {
```
Comment (alextu) [inline develocity-maven.yml:125]: I just saw a bug on the line below I'll fix in a subsequent PR (wrong default CCUD coordinates)

**Review by alextu — APPROVED** [review]
Looks good to me minus a few styling comments, TODOs on my side after this PR is merged:
- Fix the default CCUD coordinates and add unit test
- Add a smoke test on our GitLab project for injection with Develocity 1.21

File: develocity-maven.yml (hunk 1 above)
Comment (pshevche) [reply develocity-maven.yml]: Fixed in https://github.com/gradle/develocity-gitlab-templates/pull/22/commits/e52dfa902cf9a3128ad25cc95fb835ad68ad8876

File: develocity-maven.yml (hunk 2 above)
Comment (pshevche) [reply develocity-maven.yml]: Makes sense 👍 Changed in https://github.com/gradle/develocity-gitlab-templates/pull/22/commits/f4e992203dd742ab0c228696768729c2d2b60215

**Pull request merged**